fvm/meter/meter.go

```diff
@@ -1,17 +1,57 @@
 package meter
 
+type MetringOperationType uint
+
+// continuation of cadence.ComputationKind
+const (
+	// [2_000, 3_000) reserved for the FVM
+	_ = iota + 2_000
+	ComputationKindContractFunctionInvoke
+	ComputationKindHash
+	ComputationKindVerifySignature
+	ComputationKindAddAccountKey
+	ComputationKindAddEncodedAccountKey
+	ComputationKindAllocateStorageIndex
+	ComputationKindCreateAccount
+	ComputationKindEmitEvent
+	ComputationKindGenerateUUID
+	ComputationKindGetAccountAvailableBalance
+	ComputationKindGetAccountBalance
+	ComputationKindGetAccountContractCode
+	ComputationKindGetAccountContractNames
+	ComputationKindGetAccountKey
+	ComputationKindGetBlockAtHeight
+	ComputationKindGetCode
+	ComputationKindGetCurrentBlockHeight
+	ComputationKindGetProgram
+	ComputationKindGetStorageCapacity
+	ComputationKindGetStorageUsed
+	ComputationKindGetValue
+	ComputationKindRemoveAccountContractCode
+	ComputationKindResolveLocation
+	ComputationKindRevokeAccountKey
+	ComputationKindRevokeEncodedAccountKey
+	ComputationKindSetProgram
+	ComputationKindSetValue
+	ComputationKindUpdateAccountContractCode
+	ComputationKindValidatePublicKey
+	ComputationKindValueExists
+)
+
 type Meter interface {
 	// merge child funcionality
 	NewChild() Meter
 	MergeMeter(child Meter) error
 
 	// computation metering
 	MeterComputation(kind uint, intensity uint) error
+	ComputationIntensities() map[uint]uint
 	TotalComputationUsed() uint
 	TotalComputationLimit() uint
 
 	// memory metering
 	MeterMemory(kind uint, intensity uint) error
+	MemoryIntensities() map[uint]uint
```

Comment:
just a suggestion: maybe it's worth defining a type-alias for `map[uint]uint` (rather for `map[cadence.ComputationKind]uint`, when combined with my previous suggestion). seems it is being used in a lot of places